### PR TITLE
Updated DynamoSandbox.exe to avoid invalid ASM paths

### PR DESCRIPTION
**Pull request by Benglin**
I tried building Dynamo on my new machine with just `Visual Studio 2013 Ultimate` installed, Dynamo crashes because `PathManager` insists on `IPathResolver` providing valid resolution paths. Since `Autodesk Revit` isn't installed on this new machine, no ASM preloader can be found, resulting in invalid LibG folder such as `libg_0`. I have added code to prevent invalid directory such as this being added to the additional resolution paths.

Also, I realized that `DynamoIcons.csproj` requires `.NET Framework 2.0` to be installed. So I have updated the `ReadMe.md` file to include that as a build requirement.

@nguyen-binh-minh, please have a look at this one, thanks!

### CODE REVIEW

File: src/DynamoSandbox/Program.cs [hunk 1]
```diff
@@ -84,21 +88,13 @@ private static void MakeStandaloneAndRun(string commandFilePath, out DynamoViewM
             var preloaderLocation = string.Empty;
             PreloadShapeManager(ref geometryFactoryPath, ref preloaderLocation);
 
-            // TODO(PATHMANAGER): Do we really libg_xxx folder on resolution path?
-            // If not, PathResolver will be completely redundant so please remove it.
-            var pathResolver = new PathResolver(preloaderLocation);
-
-            DynamoModel.RequestMigrationStatusDialog += MigrationStatusDialogRequested;
-
```
Comment (Benglin) [inline src/DynamoSandbox/Program.cs]: The above `TODO` comment is outdated, the `PathResolver` is now mandatory because it needs to specify preloaded libraries.

File: README.md [hunk 2]
```diff
@@ -9,7 +9,15 @@ Looking to learn or download Dynamo?  Check out [dynamobim.org](http://dynamobim
 
 ## Build ###
 
-You will need Visual Studio and a git client to build Dynamo.  Find more about how to build Dynamo at our [wiki](https://github.com/DynamoDS/Dynamo/wiki).
+You will need the following to build Dynamo:
+
+- Microsoft Visual Studio 2013
+- [GitHub for Windows](https://windows.github.com/)
+- []Microsoft .NET Framework 3.5 with SP1](http://www.microsoft.com/en-sg/download/details.aspx?id=25150)
```
Comment (pboyer) [inline README.md]: Typo!  :)

File: src/DynamoSandbox/Program.cs [hunk 3]
```diff
@@ -84,21 +88,13 @@ private static void MakeStandaloneAndRun(string commandFilePath, out DynamoViewM
             var preloaderLocation = string.Empty;
             PreloadShapeManager(ref geometryFactoryPath, ref preloaderLocation);
 
-            // TODO(PATHMANAGER): Do we really libg_xxx folder on resolution path?
-            // If not, PathResolver will be completely redundant so please remove it.
-            var pathResolver = new PathResolver(preloaderLocation);
-
-            DynamoModel.RequestMigrationStatusDialog += MigrationStatusDialogRequested;
```
Comment (nguyen-binh-minh) [inline src/DynamoSandbox/Program.cs]: @Benglin why is this line deleted?

File: README.md (hunk 2 above)
Comment (Benglin) [inline README.md]: Woah, you're a markdown parser aren't you? Thanks for spotting this!

File: src/DynamoSandbox/Program.cs (hunk 3 above)
Comment (Benglin) [inline src/DynamoSandbox/Program.cs]: Good catch @nguyen-binh-minh! I obviously have deleted more than I should.

**Comment by Benglin** [comment]
Thanks again @nguyen-binh-minh and @pboyer, hopefully this time is cool.

**Comment by nguyen-binh-minh** [comment]
@Benglin LGTM

File: README.md (hunk 2 above)
Comment (pboyer) [inline README.md]: []Microsoft .NET Framework 3.5 with SP1] ->  [Microsoft .NET Framework 3.5 with SP1]
